### PR TITLE
Feature/download modal update

### PR DESCRIPTION
**Pull request by weberjavi**
This PR builds over #562 
It adds a survey each time a user wants to download some dataset (while the data explorer is not active on production this survey will get triggered when user tries to download a data set from the navigation download button).
TODO:
- [x]   Add user data to spreadsheet (follow [this gist](https://gist.github.com/qoobaa/5938bafd207e29fde46b80dc2398b9dc
) created by @qoobaa )

kapture 2018-07-09 at 9 56 21

### CODE REVIEW

**Comment by weberjavi** [comment]
@Bluesmile82 , @simaob ready to review
I've shared a google spreadsheet where data gets stored. This spreadsheet is created with testing purposes, is to be decided where this data is going to live.

**Comment by Bluesmile82** [comment]
This dropdown seems to be a bit cut at the first character. And also the scrollbar closes as in the other one. I'm beginning to think that is because they are inside a modal

image

**Comment by Bluesmile82** [comment]
I think you shouldn't be able to subscribe if you don't enter an email

File: app/javascript/app/components/modal-download/modal-download-reducers.js [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+const setModalDownloadParams = (state, { payload }) => ({
+  ...state,
+  isOpen: payload.open,
+  downloadUrl: payload.downloadUrl,
+  downloadSize: payload.size
+});
+
+const setRequiredFieldsError = (state, { payload }) => {
+  let isFieldRequiredEmpty = 0;
+  ['firstName', 'lastName', 'organization'].forEach(f => {
```
Comment (Bluesmile82) [inline app/javascript/app/components/modal-download/modal-download-reducers.js]: You could also use a some() loop

File: app/javascript/app/data/constants.js [hunk 2]
```diff
@@ -341,6 +341,9 @@ export const SOURCE_IPCC_VERSIONS = [
   { name: 'UNFCCC - AR4', source_slug: 'UNFCCC', version_slug: 'AR4' }
 ];
 
+export const USER_SURVEY_SPREADSHEET_URL =
```
Comment (Bluesmile82) [inline app/javascript/app/data/constants.js]: Let's move this to the .env

File: app/javascript/app/components/check-input/check-input-component.jsx [hunk 3]
```diff
@@ -0,0 +1,39 @@
+import React from 'react';
+import { themr } from 'react-css-themr';
+import PropTypes from 'prop-types';
+import cx from 'classnames';
+
+import styles from './check-input-styles';
+
+const CheckInputComponent = props => {
```
Comment (Bluesmile82) [inline app/javascript/app/components/check-input/check-input-component.jsx]: Cool Toggle component 😎

**Comment by simaob** [comment]
@Bluesmile82 and/or @weberjavi it feels to me that the theory of the closing dropdown on the modal, might be worth investigating.

**Comment by weberjavi** [comment]
Yes @simao, i've taken a look into it and seems it is because an automatically setted `tabindex="-1"`, i'll try to fix it on this PR

**Comment by weberjavi** [comment]
@Bluesmile82 addressed your suggestions. 
Remember to add `USER_SURVEY_SPREADSHEET_URL=https://script.google.com/macros/s/AKfycbzgN1G9IdLYO3KqlTC4gzBxR1UTX5bYXu1qRaiRn1oD9qoaq6s/exec` to env constants.

**Comment by weberjavi** [comment]
I haven't been able to reproduce the padding issue in the countries dropdown.

**Comment by simaob** [comment]
Can we close the modal after the user clicks the button?

**Comment by simao** [comment]
@weberjavi Wrong @simao please use @simaob ;)

**Comment by qoobaa** [comment]
429 Too Many Simãos

**Comment by weberjavi** [comment]
😆 sorry.
Two Tiagos and two Simaos, that's tricky (portuguese doppelganger 👯‍♂️ )

**Comment by weberjavi** [comment]
@simaob, @Bluesmile82 this PR is ready to review. I think is better to handle dropdown (on modal) bug in another PR, what do you think?

**Comment by weberjavi** [comment]
Yes @simaob, Kuba has created a [gist](https://gist.github.com/weberjavi/26082683cde640f25ca47e1f95803637) with the steps. I'll create the task on pivotal and ask Ola for them to create a spreadsheet and share the macro with us. Ok?

**Comment by simaob** [comment]
Sounds good. But for the moment we can use our own on staging.

**Comment by simaob** [comment]
Btw, I just remembered that there was a requirement that I'm not sure I wrote on the story:

- a user should only be asked for his details once every 5 days. So if a user downloads one file, it gets asked and if he fills it in, then he can download other files for the next 5 days without being prompted (as long as he is using the same browser of course).

**Comment by weberjavi** [comment]
@simaob updated with localStorage data with 5 days expiration 👆

**Comment by weberjavi** [comment]
@simaob please check the updates on jenkins and docker files 🙏

**Pull request merged**